### DISCUSSION TITLE
handling '__webpack_hmr' or 'favicon.ico' in [...slug].tsx catch all route

### DISCUSSION THREAD

**Discussion by stubar** [vercel/next.js#17471] [Help]
I have a catch all route at the root of my app `[...slug].tsx `.
It is handling requests for  `__webpack_hmr` & `favicon.ico`.
Currently for these requests I'm just returning no props in my `getServerSideProps` function.

```
  if (process.env.NODE_ENV === 'development') {
    if (['__webpack_hmr', 'favicon.ico'].includes(slug[0]))
      return { props: {} };
  }
```

This still tries to SSR the react part of the page which is of no use.
I'm also concerned the real handler for`__webpack_hmr` is not doing it's job and so HMR maybe impacted?
I understand `__webpack_hmr` is a dev only concern but `favicon.ico` requests would occur on prod presumably (I'm not in prod yet).

**Comment by dtothefp** [comment]
I also have this issue when upgrading to Webpack 5 on Next11 and using a custom Express server. @stubar did you ever find an answer?

**Reply by stubar** [reply]
I think this went away when I upgraded to a later version.  I am also on v11 and it hasn't returned for me.

**Reply by mastoj** [reply]
I experiencing the exact same thing on next13, where if it is there now if it went away when you upgraded on next11. Mainly "__webpack_hmr", I think the "favicon.ico" will get the file first. I guess the webpack stuff is implemented like some kind of middleware... shouldn't webpack deal with that request and newer pass it to the application?

I will try to return no props, but it just feels wrong.
